include/string_utils.h

```diff
@@ -380,15 +382,63 @@ std::optional<float> parse_percentage(const std::string_view s);
 //   else
 //       log_warning("%s was invalid", s.c_str());
 //
-std::optional<float> parse_prefixed_value(const char prefix, const std::string &s,
+std::optional<float> parse_prefixed_value(const char prefix, const std::string& s,
                                           const float min_value,
                                           const float max_value);
 
 // parse_prefixed_value clamped between 0 and 100
-std::optional<float> parse_prefixed_percentage(const char prefix, const std::string &s);
+std::optional<float> parse_prefixed_percentage(const char prefix,
+                                               const std::string& s);
 
 // tries to convert string to integer,
 // returns value only if succeeded
 std::optional<int> to_int(const std::string& value);
 
+#if defined(__GNUC__) || defined(__clang__)
+// Disable generic "format string is not a string literal (potentially
+// insecure)" warning on GCC/Clang. Of course, it's not a security issue for
+// us; we know what we're doing.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
+#endif
+
+template <typename... Args>
+std::string format_string(const std::string& format, Args&&... args) noexcept
+{
```

Comment:
Cheers, I wasted at least an hour on this in total and could not figure it out...

Reply:
Does not help. The pragma stays 🤷🏻 (loguru does the same thing, btw, check it out)

```
In file included from ../../tests/string_utils_tests.cpp:21:
../../include/string_utils.h:402:36: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
                std::snprintf(nullptr, 0, format, std::forward<Args>(args)...);
                                          ^~~~~~
../../tests/string_utils_tests.cpp:442:12: note: in instantiation of function template specialization 'format_string<1UL>' requested here
        EXPECT_EQ(format_string(""), "");
                  ^
../../include/string_utils.h:402:36: note: treat the string as an argument to avoid this
                std::snprintf(nullptr, 0, format, std::forward<Args>(args)...);
                                          ^
                                          "%s",
../../include/string_utils.h:417:16: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
                      format,
                      ^~~~~~
../../include/string_utils.h:417:16: note: treat the string as an argument to avoid this
                      format,
                      ^
                      "%s",
../../include/string_utils.h:402:36: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
                std::snprintf(nullptr, 0, format, std::forward<Args>(args)...);
                                          ^~~~~~
../../tests/string_utils_tests.cpp:443:13: note: in instantiation of function template specialization 'format_string<5UL>' requested here
        EXPECT_EQ(format_string("abcd"), "abcd");
                  ^
../../include/string_utils.h:402:36: note: treat the string as an argument to avoid this
                std::snprintf(nullptr, 0, format, std::forward<Args>(args)...);
                                          ^
                                          "%s",
../../include/string_utils.h:417:16: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
                      format,
                      ^~~~~~
../../include/string_utils.h:417:16: note: treat the string as an argument to avoid this
                      format,
                      ^
                      "%s",
```